src/main/java/Duke.java

```diff
@@ -1,10 +1,41 @@
 public class Duke {
+    private UserInterface ui;
+    private TaskManager tasks;
+    private FileManager fileManager;
+
+    private Duke(String pathname) {
+        this.ui = new UserInterface();
+        ui.printWelcome();
+        this.tasks = new TaskManager();
+        try{
+            this.fileManager = new FileManager();
+            if(this.fileManager.initialize(pathname, tasks)){
+                ui.printNewFile();
+            }else {
+                ui.printLoadSave();
+            }
+        }catch(DukeException e){
+            ui.printError(e.getMessage());
+        }
+    }
+
+    private void run(){
+        boolean isExit = false;
+        while(!isExit){
+            try {
+                String userCommand = ui.readLine();
+                Command c = Parser.parse(userCommand);
+                c.execute(tasks, ui, fileManager);
+                isExit = c.isExit;
```

Comment:
This might not be absolutely relevant in this context. However, what if someone has access to c? Then, s/he would be able to change `c`'s `isExit`value easily. I would suggest wrapping it as a function instead and making `isExit` `private`.

Reply:
i guess this does make sense. I'll change it next increment thank you.